client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -316,7 +433,21 @@ public void close() throws IOException {
         client.close();
     }
 
-    private static IOException addSuppressedException(IOException suppressedException, IOException currentException) {
+    private static boolean isSuccessfulResponse(String method, int statusCode) {
+        return statusCode < 300 || (HttpHead.METHOD_NAME.equals(method) && statusCode == 404);
+    }
+
+    private static boolean mayRetry(int statusCode) {
```

Comment:
I will let you and @nik9000 decide. It used to be `mustRetry`, I renamed it `mayRetry` after Nik's review :)